Common/cpp/Tools/JSLogger.cpp

```diff
@@ -0,0 +1,21 @@
+#ifdef DEBUG
+
+#include "JSLogger.h"
+#include <memory>
+
+namespace reanimated {
+JSLogger::JSLogger(const std::shared_ptr<JSRuntimeHelper> &runtimeHelper)
+    : runtimeHelper_(runtimeHelper) {}
+
+void JSLogger::warnOnJS(const std::string &warning) const {
+  runtimeHelper_->scheduleOnJS(
+      [warning = warning, &runtimeHelper_ = runtimeHelper_]() {
+        jsi::Runtime &rt = *(runtimeHelper_->rnRuntime());
+        auto console = rt.global().getPropertyAsObject(rt, "console");
+        auto warn = console.getPropertyAsFunction(rt, "warn");
+        warn.call(rt, jsi::String::createFromUtf8(rt, warning));
+      });
```

Comment:
There's no need to assign intermediate values to variables, a true JSI alpha male writes the code this way:
```suggestion
        rt.global().getPropertyAsObject(rt, "console").getPropertyAsFunction(rt, "warn").call(rt, jsi::String::createFromUtf8(rt, warning));
      });
```